tests/test_converter.py

```diff
@@ -0,0 +1,1220 @@
+            (g.RZ(0.3, 0).controlled(1).dagger().dagger(), qml.CRZ(0.3, wires=[1, 0]).inv().inv()),
+            (g.CPHASE(0.3, 0, 1), plf.ops.CPHASE(0.3, 3, wires=[0, 1])),
+            (g.CPHASE(0.3, 0, 1).dagger(), plf.ops.CPHASE(0.3, 3, wires=[0, 1]).inv()),
+            (
+                g.CPHASE(0.3, 0, 1).dagger().dagger(),
+                plf.ops.CPHASE(0.3, 3, wires=[0, 1]).inv().inv(),
+            ),
+            (g.CPHASE00(0.3, 0, 1), plf.ops.CPHASE(0.3, 0, wires=[0, 1])),
+            (g.CPHASE00(0.3, 0, 1).dagger(), plf.ops.CPHASE(0.3, 0, wires=[0, 1]).inv()),
+            (
+                g.CPHASE00(0.3, 0, 1).dagger().dagger(),
+                plf.ops.CPHASE(0.3, 0, wires=[0, 1]).inv().inv(),
+            ),
+            (g.CPHASE01(0.3, 0, 1), plf.ops.CPHASE(0.3, 1, wires=[0, 1])),
+            (g.CPHASE01(0.3, 0, 1).dagger(), plf.ops.CPHASE(0.3, 1, wires=[0, 1]).inv()),
+            (
+                g.CPHASE01(0.3, 0, 1).dagger().dagger(),
+                plf.ops.CPHASE(0.3, 1, wires=[0, 1]).inv().inv(),
+            ),
+            (g.CPHASE10(0.3, 0, 1), plf.ops.CPHASE(0.3, 2, wires=[0, 1])),
+            (g.CPHASE10(0.3, 0, 1).dagger(), plf.ops.CPHASE(0.3, 2, wires=[0, 1]).inv()),
+            (
+                g.CPHASE10(0.3, 0, 1).dagger().dagger(),
+                plf.ops.CPHASE(0.3, 2, wires=[0, 1]).inv().inv(),
+            ),
+            (g.CSWAP(0, 1, 2), qml.CSWAP(wires=[0, 1, 2])),
+            (g.CSWAP(0, 1, 2).dagger(), qml.CSWAP(wires=[0, 1, 2]).inv()),
+            (g.CSWAP(0, 1, 2).dagger().dagger(), qml.CSWAP(wires=[0, 1, 2]).inv().inv()),
+            (g.CCNOT(0, 1, 2), plf.ops.CCNOT(wires=[0, 1, 2])),
+            (g.CCNOT(0, 1, 2).dagger(), plf.ops.CCNOT(wires=[0, 1, 2]).inv()),
+            (g.CCNOT(0, 1, 2).dagger().dagger(), plf.ops.CCNOT(wires=[0, 1, 2]).inv().inv()),
+        ],
+    )
+    def test_convert_operation(self, pyquil_operation, expected_pl_operation):
+        """Test that single pyquil gates are properly converted."""
+        program = pyquil.Program()
+
+        program += pyquil_operation
+
+        with OperationRecorder() as rec:
+            load_program(program)()
+
+        assert rec.queue[0].name == expected_pl_operation.name
+        assert rec.queue[0].wires == expected_pl_operation.wires
+        assert rec.queue[0].params == expected_pl_operation.params
+
+    def test_convert_simple_program(self):
+        """Test that a simple program is properly converted."""
+        program = pyquil.Program()
+
+        program += g.H(0)
+        program += g.RZ(0.34, 1)
+        program += g.CNOT(0, 3)
+        program += g.H(2)
+        program += g.H(7)
+        program += g.X(7)
+        program += g.Y(1)
+        program += g.RZ(0.34, 1)
+
+        with OperationRecorder() as rec:
+            load_program(program)()
+
+        # The wires should be assigned as
+        # 0  1  2  3  7
+        # 0  1  2  3  4
```

Comment:
I would argue for having the behaviour described by @josh146 as default.

As an example, if I actually wanted to act on the wires outside of the template (e.g. `qml.PauliX(4)`), then it would potentially lead to unexpected behaviour.

Reply:
I agree, the current default behaviour is somewhat unexpected. I think making `wires` mandatory should be the way to go!

Reply:
I also added a property `defined_qubits` to the `ProgramLoader` so that users can inspect the qubits of the loaded program

Reply:
Cool! I'll double check that the Qiskit loader also has `wires` as mandatory, and if not, will make it so.